src/core/thread/mle_router.hpp

```diff
@@ -317,6 +317,17 @@ class MleRouter: public Mle
     Child *GetChildren(uint8_t *aNumChildren);
 
     /**
+     * This method sets the max children allowed value for this Thread interface.
+     *
+     * @param[in]  aMaxChildren  The max children allowed value.
+     *
+     * @retval  kThreadErrorNone           Successfully set the max.
+     * @retval  kThreadError_InvalidArgs   If @p aMaxChildren is not in the range [1, kMaxChildren].
+     *
+     */
+    ThreadError SetMaxAllowedChildren(uint8_t aMaxChildren);
```

Comment:
What is the expected behavior when decreasing the number of allowed children?  Is this allowed to happen while Thread protocols are enabled?  If so, how do we deal with valid children that are using an index beyond `aMaxChildren - 1`?